OpenRA.Mods.Common/Traits/Player/ProvidesPrerequisite.cs

```diff
@@ -75,8 +75,11 @@ void Update(Player owner, string faction)
 			if (info.Factions.Any())
 				enabled = info.Factions.Contains(faction);
 
+			var techTree = owner.PlayerActor.Trait<TechTree>();
 			if (info.RequiresPrerequisites.Any() && enabled)
-				enabled = owner.PlayerActor.Trait<TechTree>().HasPrerequisites(info.RequiresPrerequisites);
+				enabled = techTree.HasPrerequisites(info.RequiresPrerequisites);
+
+			techTree.Update();
```

Comment:
You may be able to save a bit of perf by only calling this if `enabled` actually changes.